データベース設計

Naoki-Hashimoto opened the pull request:
#WHAT
groupとuserのアソシエーションを「has many through」で組むのでなく、
「has_and_belongs_to_many」で組む訂正。
#WHY
中間テーブルとなるgroup_usersで特別な処理を行う必要がないため。

Commit:
create pull request

Commit:
データベース設計

#WHAT
README.mdにデータベースの概要を記載

#WHY
これから構築するデータベースの構造を明確にするため

Commit:
データベース設計

#WHAT
README.mdにデータベースの概要を記載

#WHY
これから構築するデータベースの構造を明確にするため

Commit:
データベース設計2（デフォルトの記載を削除）

#WHAT
デフォルトの記載を削除

#WHY
これから構築するデータベース構造に関する記載のみを明確にしておくため

Commit:
データベース設計3（文言統一）

#WHAT
・“group”のスペルミス → 訂正
・groups_usersとgroup_userで表記揺れ
→ ”group_users”に統一

#WHY
誤解を招く表記であったため

Commit:
データベース設計4（リレーションに関する訂正）

#WHAT
groupとuserのアソセエーションを「has many through」で組むのでなく、
「has_and_belongs_to_many」で組む訂正。
#WHY
中間テーブルとなるgroup_usersで特別な処理を行う必要がないため。

tomoki0120 commented on README.md:
```diff
@@ -1,24 +1,102 @@
+| column | type | restriction |
+|:------:|:----:|:-----------:|
+| name | string | null: false |
+| email | string | 主キー |
```
indexの指定を考えてみましょう

tomoki0120 commented on README.md:
```diff
@@ -1,24 +1,102 @@
+| column | type | restriction |
+|:------:|:----:|:-----------:|
+| group_id | integer | foreign_key: true |
+| user_id | integer | foreign_key: true |
```
reference型にしましょう！
nullについても記述しましょう！

tomoki0120 commented on README.md:
```diff
@@ -1,24 +1,102 @@
+| body | text | - |
+| image | text | - |
+| group_id | integer | foreign_key: true |
+| user_id | integer | foreign_key: true |
```
こちらも同様です！
indexの指定、null制約を考えましょう！

tomoki0120 commented on README.md:
```diff
@@ -1,24 +1,102 @@
+  group と userを繋げるテーブル。それぞれに対してbelongs
+####message
+  group ・ user、それぞれに対してbelongs
 
```
ここは
has_many:~
belongs_to:~
と書いていただいて大丈夫です！
各テーブルごとに書きましょう！

Commit:
データベース設計（indexの指定）

#WHAT
usersテーブル、massagesテーブルのカラムの一部にindexを貼った

#WHY
検索速度向上のため

Commit:
データベース設計（group_usersテーブルのカラムの型を変更）

#WHAT
group_usersテーブルのgroup_idカラム、user_idカラムの型を
integer型からreferences型に変更

#WHY
他テーブルに関連することを指すreference型を取ることで、外部キー制約をかけるため

Commit:
データベース設計（null制約の追加）

#WHAT
group_usersテーブル、messagesテーブルの外部キー（group_idカラム、user_idカラム）にnull制約を追加

#WHY
外部キーとなるカラムがnulでも保存される仕様となっていたため。

Commit:
データベース設計（リレーションの記載形式の変更）

#WHAT
group_usersテーブル、messageテーブルのリレーションに関する説明を
文章からコードへ変更

#WHY
簡略化のため

Naoki-Hashimoto commented:
データベース設計（indexの指定）以降でご指摘いただきました箇所を訂正しております。
